### PR TITLE
Add well-known timestamps to JSON for PHP

### PR DESCRIPTION
**Pull request by jackwakefield**
This adds PHP support for JSON serialization and deserialization of `google.protobuf.Timestamp` RFC 3339 timestamps.

I appreciate there may be a better way of doing this, especially to accommodate the other well-known types in the future.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by TeBoring** [comment]
Could you verify the conformance test for Timestamp has passed?
https://github.com/google/protobuf/blob/master/conformance/conformance_test.cc#L2040

**Comment by TeBoring** [comment]
Here is the script to run conformance test: https://github.com/google/protobuf/blob/master/tests.sh#L400

**Comment by TeBoring** [comment]
ok to test

**Comment by jackwakefield** [comment]
Thanks for the heads up @TeBoring, there were a number of failures but the conformance tests are now passing.

File: php/src/Google/Protobuf/Internal/Message.php [hunk 1]
```diff
@@ -1260,7 +1273,13 @@ private function fieldDataOnlyJsonByteSize($field, $value)
                 $size += 2;  // size for \"\"
                 break;
             case GPBType::MESSAGE:
-                $size += $value->jsonByteSize();
+                if ($field->isTimestamp()) {
```
Comment (TeBoring) [inline php/src/Google/Protobuf/Internal/Message.php]: Can we merge this logic to jsonByteSize?

File: php/src/Google/Protobuf/Internal/GPBWellKnown.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
+<?php
```
Comment (TeBoring) [inline php/src/Google/Protobuf/Internal/GPBWellKnown.php]: Move this to GPBUtil?

**Comment by TeBoring** [comment]
Broken travis test is for ruby. Should be unrelated.

File: php/src/Google/Protobuf/Internal/Message.php (hunk 1 above)
Comment (jackwakefield) [reply php/src/Google/Protobuf/Internal/Message.php]: That's done, I've made the same change with `serializeToJsonStream`

File: php/src/Google/Protobuf/Internal/GPBWellKnown.php (hunk 2 above)
Comment (jackwakefield) [reply php/src/Google/Protobuf/Internal/GPBWellKnown.php]: Done

**Comment by jackwakefield** [comment]
Sorry, bad rebase. Those have now been fixed.

**Comment by TeBoring** [comment]
Thanks for the help.

**Pull request merged**